### PR TITLE
[WIP] Streamable VAD

### PR DESCRIPTION
**Pull request by fpaissan**
# Contribution in a nutshell
An updated version of #1778. Closing that one to avoid excessive conflicts for merge.

# Notes for reviewing
Please also review the (private) [HF card](https://huggingface.co/speechbrain/stream-vad-crdnn-libriparty).

# Pre-review
* [x] (if applicable) add an `extra_requirements.txt` file
* [x] (if applicable) add database preparation scripts & use symlinks for nested folders (to the level of task READMEs)
* [x] (if applicable) add a recipe test entry in the depending CSV file under: tests/recipes
* [x] create a fresh testing environment (install SpeechBrain from cloned repo branch of this PR)
* [x] (if applicable) run a recipe test for each yaml/your recipe dataset
* [x] check function comments: are there docstrings w/ arguments & returns? If you're not the verbose type, put a comment every three lines of code (better: every line)
* [x] use CI locally: `pre-commit run -a` to check linters; run `pytest tests/consistency`
* [x] (optional) run `tests/.run-doctests.sh` & `tests/.run-unittests.sh`
* [ ] exhausted patience before clicking « Ready for review » in the merge box 🍄

### CODE REVIEW

**Comment by mravanelli** [comment]
I just fixed a backward incompatibility issue that was pretty problematic and caused the tests to fail. The bias of the DNN_block (used in all CRDNN) was set to false. Even though this can make sense because the bias term is already included in the batch norm part, that had an impact on the many recipes using the CRDNN. I modified the code to make the proposed change backward compatible.

**Comment by mravanelli** [comment]
Thank you for @fpaissan for this PR! This contribution is crucial as we plan to release many more streamable models for various tasks like keyword spotting, speech recognition, and possibly speech enhancement in the future.

After conducting an initial inspection, I have made the modifications above, and I have some minor comments to address:
1. We have to switch from Gdrive to Dropbox. Please, grant me access to the Gdrive folder so that I can download the output folder and upload it to the Speechbrain Dropbox.
2. The header of `inference.py` could be improved. I suggest adding instructions on how to run the code with additional details similar to what you provided in the HF repository.
3. I propose creating a short video (around 2-3 minutes) that demonstrates the process of performing streamable inference. This would be valuable for users trying to understand the implementation better.

Feel free to take care of the previous comments. My next step will be to run the training and inference scripts along with the recipe test.

**Comment by mravanelli** [comment]
I did some fixes and now training is running on my side. My next step is to check the streamable inference part.

**Comment by mravanelli** [comment]
https://huggingface.co/speechbrain/stream-vad-crdnn-libriparty/blob/main/README.md => It looks like the inference.py script only works with Apple devices, due to a limitation in torchaudio (in fact, it doesn't work on my Linux machine). If so, we should mention that here as well.
